### PR TITLE
[Term Entry] Java ArrayList .listiterator() #4341

### LINKED ISSUE #4341: [Term Entry] Java ArrayList .listiterator()
### Reason/inspiration (optional)

Please add the `.listiterator()` method as a new term entry for [ArrayList](https://www.codecademy.com/resources/docs/java/array-list)�. Provide the entry in a markdown file and place under the file path `content/java/concepts/array-list/terms/listiterator/listiterator.md`

The entry should have the following:
�	An introductory paragraph about what the method is and does.
�	A syntax section with a pseudo code block with an explanation of it.
�	An example showing the method in use.

Please refer to the [term entry templates](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by selenahuang017**
### Description

Adds an additional term entry for Java ArrayList listiterator under the file path content/java/concepts/array-list/terms/listiterator/listiterator.md

### Issue Solved

Addresses: [Term Entry] Java ArrayList .listiterator() #4341 [https://github.com/Codecademy/docs/issues/4341](url)
Closes #4341 

### Type of Change

- Adding a new entry

### Checklist

- [ x] All writings are my own.
- [ x] My entry follows the Codecademy Docs style guide.
- [ x] My changes generate no new warnings.
- [ x] I have performed a self-review of my own writing and code.
- [ x] I have checked my entry and corrected any misspellings.
- [ x] I have made corresponding changes to the documentation if needed.
- [ x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by letsbuildanything** [comment]
@selenahuang017 @selena-huang
Please sign the CLA using both accounts. We can begin the review process once this is completed.

**Comment by selenahuang017** [comment]
I've signed them :)

**Comment by selenahuang017** [comment]
@letsbuildanything @mamtawardhani just checking on the status of this, is there anything else I need to do?

File: content/java/concepts/array-list/terms/listiterator/listiterator.md
Comment (letsbuildanything) [inline content/java/concepts/array-list/terms/listiterator/listiterator.md:84]:
```suggestion
    ListIterator<String> it = arrayList.listIterator(2);
```

File: content/java/concepts/array-list/terms/listiterator/listiterator.md
Comment (letsbuildanything) [inline content/java/concepts/array-list/terms/listiterator/listiterator.md:87]:
```suggestion
    while (it.hasNext()) {
```

File: content/java/concepts/array-list/terms/listiterator/listiterator.md
Comment (letsbuildanything) [inline content/java/concepts/array-list/terms/listiterator/listiterator.md:119]:
```suggestion
    System.out.println("ArrayList size: " + arrayList.size());
```

File: content/java/concepts/array-list/terms/listiterator/listiterator.md
Comment (letsbuildanything) [inline content/java/concepts/array-list/terms/listiterator/listiterator.md:123]:
```suggestion
        ListIterator<String> it = arrayList.listIterator(5);
```

File: content/java/concepts/array-list/terms/listiterator/listiterator.md
Comment (letsbuildanything) [inline content/java/concepts/array-list/terms/listiterator/listiterator.md:135]:
```suggestion
ArrayList size: 4
Exception: java.lang.IndexOutOfBoundsException: Index: 5, Size: 4
```

**Review by letsbuildanything — CHANGES_REQUESTED** [review]
@selenahuang017 Thank you for contributing to the Codecademy Docs!
I've requested some changes. Please take a moment to address them. If you have any concerns or reasons to reject specific suggestions, feel free to discuss them.

**Review by selenahuang017 — COMMENTED** [review]
Updating java listiterator term as per feedback.
Edit: oops I didn't realize this was just a comment, I have updated the commit

**Review by letsbuildanything — APPROVED** [review]
@selenahuang017 Thank you for making the changes.
Ready for the second round of review!

**Comment by selenahuang017** [comment]
just checking in here, it looks like it needs maintainer approval @mamtawardhani , is there anyone I should reach out to?

**Comment by mamtawardhani** [comment]
> just checking in here, it looks like it needs maintainer approval @mamtawardhani , is there anyone I should reach out to?

Hey @selenahuang017, this entry is in a second round of review, it will be done ASAP, don't worry.

File: content/java/concepts/array-list/terms/listiterator/listiterator.md
Comment (ishg-153) [inline content/java/concepts/array-list/terms/listiterator/listiterator.md:3]:
```suggestion
Description: 'Iterates over the elements in an ArrayList in both directions.'
```

**Review by ishg-153 — CHANGES_REQUESTED** [review]
@selenahuang017 Great entry! Please make this one change. Let me know once you are done.

**Comment by mamtawardhani** [comment]
@ishg-153 seems like the changes are done, you can check and merge this

**Review by ishg-153 — APPROVED** [review]
@selenahuang017 Great! This is ready to merge.

**Pull request merged**
